sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_methods.c

```diff
@@ -273,6 +274,51 @@ static void test_az_iot_hub_client_methods_response_get_publish_topic_succeed(vo
   assert_int_equal(sizeof(expected_topic) - 1, test_length);
 }
 
+static void test_az_iot_hub_client_methods_response_get_publish_topic_user_status_succeed(
+    void** state)
+{
+  (void)state;
+
+  char test_buf[TEST_SPAN_BUFFER_SIZE];
+  size_t test_length;
+
+  az_iot_hub_client client;
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, NULL) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 12345;
+  const char expected_topic[] = "$iothub/methods/res/12345/?$rid=2";
+
+  assert_true(
+      az_iot_hub_client_methods_response_get_publish_topic(
+          &client, request_id, status, test_buf, sizeof(test_buf), &test_length)
+      == AZ_OK);
+
+  assert_string_equal(expected_topic, test_buf);
+  assert_int_equal(sizeof(expected_topic) - 1, test_length);
+}
+
+static void test_az_iot_hub_client_methods_response_get_publish_topic_user_status_small_buf_fail(
+    void** state)
+{
+  (void)state;
+
+  az_iot_hub_client client;
+  assert_true(az_iot_hub_client_init(&client, test_device_hostname, test_device_id, NULL) == AZ_OK);
+
+  az_span request_id = AZ_SPAN_LITERAL_FROM_STR("2");
+  uint16_t status = 12345;
```

Comment:
To properly test the range I would suggest changing this to UINT16_MAX (65535).

Reply:
updated